typescript/infra/config/environments/mainnet3/warp/configGetters/getEverclearConfig.ts

```diff
@@ -0,0 +1,341 @@
+  linea: '0xa219439258ca9da29e9cc4ce5596924745e12b93',
+  scroll: '0xf55BEC9cafDbE8730f096Aa55dad6D22d44099Df',
+  zircuit: '0x46dDa6a5a559d861c06EC9a95Fb395f5C3Db0742',
+  mode: '0xf0F161fDA2712DB8b566946122a5af183995e2eD',
+  base: '0xfde4C96c8593536E31F229EA8f37b2ADa2699bb2',
+};
+
+// Ownership configuration (following CCTP pattern)
+const owners: Record<EverclearChain, string> = {
+  ethereum: awSafes['ethereum'],
+  bsc: awSafes['bsc'],
+  optimism: awSafes['optimism'],
+  zksync: awSafes['zksync'],
+  arbitrum: awSafes['arbitrum'],
+  polygon: awSafes['polygon'],
+  avalanche: awSafes['avalanche'],
+  unichain: awSafes['unichain'],
+  sonic: awSafes['sonic'],
+  mantle: awSafes['mantle'],
+  ink: awSafes['ink'],
+  linea: awSafes['linea'],
+  scroll: awSafes['scroll'],
+  zircuit: awSafes['zircuit'],
+  mode: awSafes['mode'],
+  base: awSafes['base'],
+};
+
+// Chain ID to name mapping for signatures.json parsing
+const CHAIN_ID_TO_NAME: Record<string, EverclearChain> = {
+  '1': 'ethereum',
+  '56': 'bsc',
+  '10': 'optimism',
+  '324': 'zksync',
+  '42161': 'arbitrum',
+  '137': 'polygon',
+  '43114': 'avalanche',
+  '130': 'unichain',
+  '146': 'sonic',
+  '5000': 'mantle',
+  '57073': 'ink',
+  '59144': 'linea',
+  '534352': 'scroll',
+  '48900': 'zircuit',
+  '34443': 'mode',
+  '8453': 'base',
+};
+
+interface FeeData {
+  sig: string;
+  deadline: string;
+  tokenFee: string;
+  nativeFee: string;
+}
+
+type EverclearAsset = 'USDC' | 'WETH' | 'USDT';
+
+// Function to fetch fee data from signatures.json for a given asset
+function fetchFeeData(asset: EverclearAsset) {
+  const dirname = path.dirname(fileURLToPath(import.meta.url));
+  const signaturesPath = path.join(dirname, '../../../../../signatures.json');
+  const signaturesData = JSON.parse(fs.readFileSync(signaturesPath, 'utf8'));
+
+  const result: Partial<
+    Record<EverclearChain, Partial<Record<EverclearChain, FeeData>>>
+  > = {};
+
+  if (!signaturesData.signatures?.[asset]) {
+    throw new Error(`No fee data found for ${asset} in signatures json`);
+  }
+
+  for (const [originChainId, destinations] of Object.entries(
+    signaturesData.signatures[asset],
+  )) {
+    const originName = CHAIN_ID_TO_NAME[originChainId];
+    if (!originName) continue;
+    result[originName] = {};
+
+    for (const [destChainId, feeInfo] of Object.entries(
+      destinations as Record<string, any>,
+    )) {
+      const destName = CHAIN_ID_TO_NAME[destChainId];
+      if (!destName) continue;
+
+      result[originName][destName] = {
+        ...feeInfo,
+      };
+    }
+  }
+
+  return result;
+}
+
+const getConfigFromFeeData = (
+  asset: EverclearAsset,
+  routerConfig: ChainMap<RouterConfigWithoutOwner>,
+): ChainMap<HypTokenRouterConfig> => {
+  const feeDataByChain = fetchFeeData(asset);
+  const tokenAddresses =
+    asset === 'USDC'
+      ? usdcTokenAddresses
+      : asset === 'WETH'
+        ? wethTokenAddresses
+        : usdtTokenAddresses;
```

Comment:
a constant mapping might be more readable and extensible to if we add more assets

Reply:
Added a switch statement